manage.py

```diff
@@ -37,6 +37,32 @@ def add_event_identifier():
         save_to_db(event)
 
 
+@manager.command
+def fix_event_and_speaker_images():
+    events = Event.query.filter(Event.original_image_url != None,
+                                or_(Event.thumbnail_image_url == None, Event.large_image_url == None,
+                                    Event.icon_image_url == None)).all()
+    from app.api.helpers.files import create_save_image_sizes
+    for event in events:
+        resized_images = create_save_image_sizes(event.original_image_url, 'event-image', event.id)
+        event.large_image_url = resized_images['large_image_url']
+        event.thumbnail_image_url = resized_images['thumbnail_image_url']
+        event.icon_image_url = resized_images['icon_image_url']
+        db.session.add(event)
```

Comment:
This should be done automatically. You're repeating code. See how it is already done

Reply:
The function returns a dictionary of these images, also similar way the celery task of resizing image is done, the function just save the image to directory but we have to save to db. It will not be done automatically.


https://github.com/fossasia/open-event-server/blob/e8b367f7955bf4b90fb79ff7514510840306ec9d/app/api/helpers/tasks.py#L146-L154

Reply:
Point me where I am wrong. I will correct it

Reply:
Then you should call the function calling this function, and not repeat the functionality of assigning resized images and saving the event and speaker

Reply:
Then how will the changes go to db?

Reply:
How is it done when you send a request to set the image using the API?

Reply:
We just hit the api and the images are stored in the folder, my point is 
when we had  a check that  
`Event.thumbnail_image_url == None, Event.large_image_url == None,
                                     Event.icon_image_url == None)`
that means that the resizing has not occured and the location of resized-image is not present in the db.

What I m understanding that you are saying to just generate the reisized-image and not update the db. Correct?

But then still, the db column values of the above checks still have null values.

Reply:
Also I debugged earlier that calling  create_save_image_sizes just generates the resized image but doens't update the DB.

Reply:
Then your previously merged PR is incomplete, you said resized image URLs were being saved in the DB. Should I revert the PR?

Reply:
>
How is it done when you send a request to set the image using the API?

For patching the event image, again same task is done by calling a celery tasks which i mentioned above

Reply:
>Then your previously merged PR is incomplete, you said resized image URLs were being saved in the DB. Should I revert the PR?

No no, the celery tasks will do that I mentioned above, check
https://github.com/fossasia/open-event-server/blob/e8b367f7955bf4b90fb79ff7514510840306ec9d/app/api/helpers/tasks.py#L128-L139

Reply:
So, why aren't you calling this method? Why are you duplicating the code?
https://github.com/fossasia/open-event-server/pull/6645#discussion_r355121074

Reply:
>Then your previously merged PR is incomplete, you said resized image URLs were being saved in the DB. Should I revert the PR?

When you create an event, they will invoke the celery task of resize_event_images_task which will call `create_save_image_sizes` to generate the resized-image and the celery tasks will save them to DB.

Reply:
Because the celery again queries-for the event, that's why. 2 query for fetching the same event. What shall I do now?

Reply:
Call `resize_event_images_task`

Reply:
>#6645 (comment)

Maybe if you named the function, it would not have confused. `Calling this function is `ambigous for me.

Reply:
That's called spoonfeeding. I don't do that. These are the things you should realize and learn by your own. It's not OK to repeat code. It would have taken me 5 minutes to just suggest the exact changes to get the job done, but that is not my goal